Ajustes no renderer e em models.to_dict(), nos dados de relacionamento

robertatakenaka opened the pull request:
Ajustados to_dict()  para Attempt, ArticlePkg, Ticket.

Ticket: adicionado comments com uma lista de  {author, message, date}.
Attempt: adicionado checkin e validation como {started_at: date, finished_at:date, notices: [{message:'', label: '', status:'', date:''}, ]
ArticlePkg: adicionado tickets.

Foram alterados no renderer: 
1. add_meta mudou de nome para format_response, por ser mais adequado ao conteúdo do método, pois nem sempre add_meta adiconava meta. 
2. o conteúdo de add_resource_uri foi modificado de modo que ele adiciona resource uri apenas para um objeto e, não mais, para um objeto e uma lista deles.
3. testes ajustados para 1 e 2

robertatakenaka committed:
ArticlePkg: tickets and attempts are displayed as list of resource uri
Tickets: comments as a list of {author, message, date}.
Changed add_meta to format_response. Because it formats the response: adding resource uri, but only if there is a list of objects, it also adds meta.
Changed content of add_resource_uri. Previously it added resouces uri to one object or to a list of objects, but now it adds resources uri to an object.
Adjustments in test_renderer

robertatakenaka committed:
appending checkin and validations (which are Checkpoint) to Attempt.

robertatakenaka committed:
Adding checkin and validation to Attempt.to_dict(). #133

robertatakenaka committed:
Clean unnecessary comments

gustavofonseca commented on balaio/models.py:
```diff
@@ -62,7 +63,10 @@ def to_dict(self):
                     filepath=self.filepath,
-                    is_valid=self.is_valid)
+                    is_valid=self.is_valid,
+                    checkin=next(_checkpoint.to_dict() for _checkpoint in self.checkpoint if _checkpoint.point == Point.checkin),
```
Esse código não está mto legal. Os checkpoints são percorridos mais de 1 vez, e a invocação do _next_ demonstra que só o primeiro ítem é utilizado. 

``` python
checkpoints = {cp.point.name:cp.to_dict() for cp in self.checkpoint if cp.point is not Point.checkout}
```

Depois vc pode mesclar o dict _checkpoints_ com o outro dict por meio do método _.update_ por exemplo.

robertatakenaka replied:
ok

gustavofonseca commented on balaio/models.py:
```diff
@@ -222,7 +232,8 @@ class Checkpoint(Base):
     attempt = relationship('Attempt',
-                           backref=backref('checkpoint'))
+                            backref=backref('checkpoint'))
+                           #backref=backref('checkpoint'))
```
Comentário esquecido

robertatakenaka committed:
Ajustes de acordo com comentarios de revisão

robertatakenaka commented:
Ajustes feitos

robertatakenaka committed:
Merge branch 'master' of git://github.com/scieloorg/balaio into 133_related_resources

Conflicts:
	balaio/renderers.py
	balaio/tests/test_renderer.py

robertatakenaka committed:
total_count to total in docs.

jamilatta commented on balaio/renderers.py:
```diff
@@ -39,17 +30,16 @@ def add_meta(self, value):
                 'offset': value['offset'],
```
O param value não existe mais.

jamilatta replied:
Na verdade é necessário criar testes utilizando esses params.

robertatakenaka committed:
Fixes tests which were not OK

robertatakenaka commented:
Corrigido. Por favor, reavaliar.

gustavofonseca commented on balaio/models.py:
```diff
@@ -210,6 +213,13 @@ def status(self):
+    def to_dict(self):
+        return dict(label=self.label,
+                    message=self.message,
+                    status=str(self.status).replace('Status.', ''),
```
`status=self.status.name`

robertatakenaka committed:
Troca de _N() por _positive_int_or_zero() (#142)
Merge branch 'master' of git://github.com/scieloorg/balaio into 133_related_resources

Conflicts:
	balaio/gateway_server.py
	balaio/renderers.py
	balaio/tests/test_renderer.py
